core/org.openjdk.jmc.flightrecorder/src/main/java/org/openjdk/jmc/flightrecorder/internal/parser/v1/TypeManager.java

```diff
@@ -469,6 +475,14 @@ private IValueReader createFieldReader(FieldElement f, String valueType) throws
 		return f.isArray() ? new ArrayReader(reader, header) : reader;
 	}
 
+	private static void skipEvent(int size, IDataInput input) throws IOException {
+		// We've read the size of the event (int) and the type (long) so far
+		for (int i = 0; i < size - DataInputToolkit.INTEGER_SIZE - DataInputToolkit.LONG_SIZE; i++) {
```

Comment:
The size and event type ID field is likely compressed, so using a fixed constant size will not work.  I think you need to find the offset of the first byte in the size field and then add the size to find next event.

Reply:
Ugh. You're right. And what's even more fun, I don't need to do anything at all. The event read loop is already handling it by seeking to the absolute position for the next event. Jumping to the next event is a noop - nothing needs be done.

Reply:
Any chance you have a recording with a few events that has metadata, followed by an event with no metadata, then again followed by a few events with metadata available? :)

Reply:
No, it should not happen.

Reply:
Ok, then we should be all good.